code/crates/sync/src/scoring/ema.rs

```diff
@@ -0,0 +1,107 @@
+use std::time::Duration;
+
+use malachitebft_peer::PeerId;
+
+use super::{Score, ScoringStrategy, SyncResult};
+
+/// Exponential Moving Average scoring strategy
+#[derive(Copy, Clone, Debug)]
+pub struct ExponentialMovingAverage {
+    /// Learning rate for successful responses
+    pub alpha_success: f64,
+
+    /// Learning rate for timeouts
+    pub alpha_timeout: f64,
+
+    /// Learning rate for failures
+    pub alpha_failure: f64,
+
+    /// Threshold for slow responses.
+    ///
+    /// This should typically be smaller than both the expected
+    /// block time and the sync request timeout, as we do not
+    /// want responses that are slower than the expected block
+    /// time to be considered successful otherwise a node might
+    /// not be able to keep up with the network.
+    pub slow_threshold: Duration,
+}
+
+impl Default for ExponentialMovingAverage {
+    fn default() -> Self {
+        Self::new(
+            0.2,                    // Success
+            0.1,                    // Timeout
+            0.15,                   // Failure
+            Duration::from_secs(1), // Slow threshold
```

Comment:
Isn't one second too little to reset the scores? We would be giving bad peers another chance too fast I think. 

Of course this will also depend on the application. For instance when values are too big, they will take more time to transmit, so the app developer should take this into account (though then the block time and sync timeout will also be bigger).

A slightly different strategy would be to reset scores only when there are not enough good peers available. (No need to address this in this PR.)